### ISSUE TITLE
external_reply.origin.sender_user does not show the correct user (replying to a forward)

### ISSUE THREAD

**Issue by wiz0u** [tdlib/telegram-bot-api#659]
When replying to a message from another chat which was a forward, the `sender_user` seems to be filled with information from the initially forwarded user, rather than information about the message sender in the other chat:

in the below example, replying to https://t.me/iamnotstalin2Chat/665,
we would expect sender_user to identify Андрей rather than НЕ СТАЛИН

![image](https://github.com/user-attachments/assets/14ac8d67-6f82-4f5a-81e3-0b4c72201dd8)

```json
{
  "update_id": 144536348,
  "message": {
    "message_id": 1388580,
    "from": {
      "id": 301455566,
      "is_bot": false,
      "first_name": "Wizou",
      "username": "Wiz0u",
      "language_code": "fr",
      "is_premium": true
    },
    "chat": {
      "id": 301455566,
      "first_name": "Wizou",
      "username": "Wiz0u",
      "type": "private"
    },
    "date": 1730741106,
    "external_reply": {
      "origin": {
        "type": "user",
        "sender_user": {
          "id": 6802950856,
          "is_bot": false,
          "first_name": "НЕ СТАЛИН",
          "username": "iamnotstalin_person"
        },
        "date": 1730736974
      },
      "chat": {
        "id": -1002118288285,
        "title": "Вспомогательный чат 2",
        "username": "iamnotstalin2Chat",
        "type": "supergroup"
      },
      "message_id": 665,
      "sticker": {
        "width": 512,
        "height": 512,
        "emoji": "😍",
        "is_animated": true,
        "is_video": false,
        "type": "regular",
        "thumbnail": {
          "file_id": "AAMCAgADGQEAARUwJGcpA3KFuMKLCvxuKbniUEU3cKkPAAJ7AAPBnGAM2xofth1UNogBAAdtAAM2BA",
          "file_unique_id": "AQADewADwZxgDHI",
          "file_size": 4580,
          "width": 128,
          "height": 128
        },
        "thumb": {
          "file_id": "AAMCAgADGQEAARUwJGcpA3KFuMKLCvxuKbniUEU3cKkPAAJ7AAPBnGAM2xofth1UNogBAAdtAAM2BA",
          "file_unique_id": "AQADewADwZxgDHI",
          "file_size": 4580,
          "width": 128,
          "height": 128
        },
        "file_id": "CAACAgIAAxkBAAEVMCRnKQNyhbjCiwr8bim54lBFN3CpDwACewADwZxgDNsaH7YdVDaINgQ",
        "file_unique_id": "AgADewADwZxgDA",
        "file_size": 37975
      }
    },
    "text": "dump"
  }
}
```

**Comment by wiz0u** [comment]
maybe external_reply structure should have both a `origin` field AND a `forward_origin` field, to differenciate the message sender and the forwarded user

**Comment by levlam** [comment]
This is intended. [MessageOrigin](https://core.telegram.org/bots/api#messageorigin) always represents the initial sender of the message both in forwards and `ExternalReplyInfo`.

Actual source of the forward is available only in Saved Messages, the Replies chat, which aren't relevant for bots.

If you want Telegram to provide the latter for all chats, then you need to create a card at https://suggestions.telegram.org, unless one already exists.

**Comment by levlam** [comment]
> why not add a from (User) field to ExternalReplyInfo ?

Because there is no such information as I explained.

> seeing the screenshot above (taken in chat with bot), one would expect that information to be available (to bots too)

Is it Telegram Desktop? You must restart it to see the actual message.

**Comment by wiz0u** [comment]
> Is it Telegram Desktop? You must restart it to see the actual message.

yes. I just restarted Telegram Desktop and went straight to my chat with @JsonDumpBot,
the message still appeared the same with all information about real sender.

note: I'm not even a member of https://t.me/iamnotstalin2Chat so Telegram Desktop did not fetch the origin message from my chats.

**Comment by wiz0u** [comment]
in Telegram Android, the message appear as you said, with only НЕ СТАЛИН

**Comment by levlam** [comment]
> I just restarted Telegram Desktop and went straight to my chat with @JsonDumpBot,

You didn't restart it. You closed the window and minimized the app to tray. You can close it from there.

**Comment by wiz0u** [comment]
Trust me, i closed it. (retried again now, same)
I disabled minimize to tray

**Comment by levlam** [comment]
There is no way Telegram Desktop can keep this state of the message if closed.

**Comment by wiz0u** [comment]
🤷🏻‍♂️ maybe it auto-fetches the message to get more info?
https://t.me/iamnotstalin2Chat is a public chat after all
